### ISSUE TITLE
Rewrite the UserViewModelBinder registration

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#20]
### Who

- [ ] Anonymous visitor
- [ ] Patient
- [ ] Doctor
- [ ] Content editor
- [ ] Marketer
- [ ] Administrator
- [x] Developer

### Needs what

To register the custom model binder in different ways than it has been done in MVC 5. The [model binder provider](https://docs.microsoft.com/en-us/aspnet/core/mvc/advanced/custom-model-binding?view=aspnetcore-3.1#implementing-a-modelbinderprovider) is a good fit, however, might not work reliably.

### Why

The user model is passed along as an interface in MedioClinic. Thus it needs a custom binder that casts it into a proper class, based on what the role of the current user is.

### Issue expands upon

This issue expands upon another issue(s): #10 

### Blockers

This issue is blocked by the following issue(s): #`issue number`
